platform/cog-platform-fdo.c

```diff
@@ -1266,6 +1392,130 @@ on_export_fdo_egl_image(void *data, struct wpe_fdo_egl_exported_image *image)
     wl_surface_commit (win_data.wl_surface);
 }
 
+#if COG_ENABLE_WESTON_DIRECT_DISPLAY
+static void
+create_succeeded(void *data,
+		 struct zwp_linux_buffer_params_v1 *params,
+		 struct wl_buffer *new_buffer)
+{
+	zwp_linux_buffer_params_v1_destroy (params);
+
+	struct buffer *buffer = data;
+	buffer->buffer = new_buffer;
+}
+
+static void
+create_failed(void *data, struct zwp_linux_buffer_params_v1 *params)
+{
+    struct buffer *buffer = data;
+
+    buffer->buffer = NULL;
+    zwp_linux_buffer_params_v1_destroy (params);
+    g_assert_not_reached ();
```

Comment:
This looks odd. Is it really expectated that `create_failed()` will never be called? If that's the case, why not asserting at the top of the function and removing the rest of the code?

Reply:
I think the assert could be removed actually.